ok-marketplace-be-mappers-openapi/src/main/kotlin/com/polyakovworkbox/tasktracker/backend/common/mapping/ContextToTransport.kt

```diff
@@ -0,0 +1,91 @@
+package com.polyakovworkbox.tasktracker.backend.common.mapping
+
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.ApiError
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.CreateTaskResponse
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.DeleteTaskResponse
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.Measurability
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.ReadTaskResponse
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.ResponseResult
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.UpdatableTask
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.UpdateTaskResponse
+import com.polyakovworkbox.otuskotlin.tasktracker.transport.openapi.task.models.SearchTasksResponse
+import com.polyakovworkbox.tasktracker.backend.common.context.BeContext
+import com.polyakovworkbox.tasktracker.backend.common.models.task.TaskIdReference
+import com.polyakovworkbox.tasktracker.backend.common.models.general.ApiError as DomainApiError
+import com.polyakovworkbox.tasktracker.backend.common.models.task.Task as TaskDomain
+import com.polyakovworkbox.tasktracker.backend.common.models.task.Measurability as MeasurabilityDomain
+
+
+fun BeContext.toCreateResponse() = CreateTaskResponse(
+    responseId = this.responseId.id,
+    messageType = CreateTaskResponse::class.java.simpleName,
+    result = if(errors.isEmpty()) ResponseResult.SUCCESS else ResponseResult.ERROR,
+    errors = this.errors.mapErrorsToTransport(),
+    task = if(errors.isEmpty()) this.responseTask.mapToTransport() else null
```

Comment:
Все то же самое можно переписать более кратно 
```suggestion
    task = responseTask.takeIf { errors.isEmpty() }?.mapToTransport()
```